frontend/rolecall/src/app/cast/cast-drag-and-drop.component.ts

```diff
@@ -127,206 +153,150 @@ export class CastDragAndDrop implements OnInit {
-    let tempData = filledPoses.map((val, posInd) => {
-      let colObs = val.groups.sort((a, b) => a.group_index < b.group_index ? -1 : 1);
-      let maxInd = this.getMaxDancerIndexForCast(posInd);
-      let subcastStrs: string[][] = [];
-      for (let i = 0; i < maxInd; i++) {
-        subcastStrs.push([]);
       }
-      for (let col of colObs) {
-        for (let i = 0; i < col.members.length; i++) {
-          let member = col.members[i];
-          subcastStrs[member.position_number].push(member.uuid);
+    }
+
+    // Sort positions by position order
+    this.castPositions = this.castPositions.sort((a, b) => a.pos.order < b.pos.order ? -1 : 1 );
+    // Sort positions inside cast
+    this.cast.filled_positions = this.cast.filled_positions
+    .filter(val => this.castPositions.find(castPos => castPos.pos.uuid == val.position_uuid))
+    .sort((a, b) => {
+      const castPositionOrderA = this.castPositions.find(castPos => castPos.pos.uuid == a.position_uuid).pos.order;
+      const castPositionOrderB = this.castPositions.find(castPos => castPos.pos.uuid == b.position_uuid).pos.order;
+      return castPositionOrderA < castPositionOrderB ? -1 : 1
+    });
+
+    for (let posIx = 0; posIx < this.cast.filled_positions.length; posIx++) {
+      const filledPos = this.cast.filled_positions[posIx];
+      const uiPos = this.castPositions[posIx];
+      let maxDancerIx = 0;
+      for (let groupIx = 0; groupIx < filledPos.groups.length; groupIx++) {
+        const group = filledPos.groups[groupIx];
+        const group_index = group.group_index;
+        for (let memberIx = 0; memberIx < group.members.length; memberIx++) {
+          const member = group.members[memberIx];
+          if (member.position_number >= maxDancerIx) {
+            maxDancerIx = member.position_number;
+            if (maxDancerIx >= uiPos.castRows.length) {
+              uiPos.castRows.push({
+                subCastDancers: new Array(CAST_COUNT),
+              });
+            }
+          }
+          const dancer = this.userAPI.users.get(member.uuid);
+          uiPos.castRows[member.position_number].subCastDancers[group_index] = {
+            uuid: dancer.uuid,
+            firstName: dancer.first_name,
+            lastName: dancer.last_name,
+            email: dancer.contact_info.email,
+          };
         }
       }
-      let subcasts = subcastStrs.map(val => val.map(val2 => this.userAPI.users.get(val2)));
-      return subcasts;
-    });
-    this.data = tempData;
-    for (let posInd = 0; posInd < this.positionVals.length; posInd++) {
-      this.setColumnHeaders(posInd);
-      this.ensureEmptyArrayAtEnd(posInd);
-      this.updateEmptyRows(posInd);
+      uiPos.castSize = Math.max(maxDancerIx + 1, uiPos.pos.size);
     }
   }
 
   drop(event: CdkDragDrop<User[]>) {
+    const fromContainer = event.previousContainer.id;
+    const toContainer = event.container.id;
+    const fromIxs = fromContainer.split(':');
```

Comment:
Please fix suffixes

Reply:
Which suffixes? (I am not being funny, I don't know.)

Reply:
Was referring to `*Ixs` and `*Ix`